### ISSUE TITLE
Scanning 1D barcodes on iPhone's

### ISSUE THREAD

**Issue by tomurphy-78** [mebjas/html5-qrcode#423]
Hi 

I am trying to scan 1D barcodes, using the demo page.  On android devices using chrome, it scans really well.  It also scans well on older iPhones using safari.  However on newer iPhones using both safari and chrome, it rarely scans the 1D barcode and sometimes not at all, it looks like the camera is having trouble recognising it??

Have you or anyone seen this behaviour?  And have been able to fix it?

I also posted on your blog before seeing this page, so apologies for the double post.

Cheers
Tom

**Comment by mebjas** [comment]
You are right, it seems to work on Pixel but doesn't scan on latest iphone - I'll look into this, thanks for posting.

**Comment by iamsrujal** [comment]
I have checked with iphone 13 pro with IOS version 15.3.1.

Used this website to generate 1d barcode [https://poscatch.com/barcode-generator/](url)

Barcode scanning works perfectly fine for me.

**Comment by alexander-rieder** [comment]
for me Barcode Scanning only works on the pc using a webcam or by file upload (also only on pc)

on iOS and Android I cant scan the barcode... 

I attached the "problematic" barcode

<img width="343" alt="itf_barcode" src="https://user-images.githubusercontent.com/46959878/155296163-b78102eb-0d6a-4bb9-af1f-778f92b27c67.PNG">

**Comment by ROBERT-MCDOWELL** [comment]
I think the issue is the black touching the bar code at the top.
with a conventional infrared scanner it's not a problem since the field scanned is an horizontal line.
with a camera it's more complex, it depends to the camera quality and focus.
did you try with another bar code without this black top?
